tests/test_types.py

```diff
@@ -1699,6 +1699,10 @@ class Foobar(BaseModel):
     with pytest.warns(DeprecationWarning, match=r'`secret_str.display\(\)` is deprecated'):
         assert f.empty_password.display() == ''
 
+    # Assert that SecretStr is equal to SecretStr if the secret is the same.
+    assert f == f.copy()
+    assert f != f.copy(update=dict(password='4321'))
```

Comment:
this is fine, but please add an explicit test, e.g.:

```py
def test_secret_str():
    assert SecretStr('123') == SecretStr('123')
    assert SecretStr('123') != SecretStr('321')
    assert SecretStr('123') != '123'
    assert SecretStr('123') is not SecretStr('123')
```
And the same for `SecretBytes`.

Reply:
I kept the copy version and added these below. Ok?